src/v/ssx/thread_worker.h

```diff
@@ -237,7 +319,52 @@ class thread_worker {
 
 private:
     ss::sharded<ss::gate> _gate;
-    impl::thread_worker _impl;
+    thread_worker_config _config;
+    impl::thread_worker _impl = impl::thread_worker(&_config);
+};
+
+/**
+ * sharded_thread_worker runs tasks in a std::thread, and creates a thread core
+ * each reactor core so that there is a twin std::thread for each reactor core.
+ *
+ * By running in a std::thread, it's possible to make blocking calls such as
+ * file I/O and posix thread primitives without blocking a reactor.
+ *
+ * The thread worker will drain all operations before joining the thread in
+ * stop(), but it should be noted that joining a thread may block. As such, this
+ * class is most suited to run for the lifetime of an application, rather than
+ * short-lived.
+ */
+class sharded_thread_worker {
+public:
+    explicit sharded_thread_worker(thread_worker_config config)
```

Comment:
I think it would be cleaner to wire config through `start()`, rather than at construction.

Reply:
Yupp much better :)